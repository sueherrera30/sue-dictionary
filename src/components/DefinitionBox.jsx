import styles from './styles/definitionBox.module.css';

const DefinitionBox = ({ meanings, partOfSpeech, phonetic }) => (
    <>
        <section className={styles.mainContainer}>
        <p className={styles.speech}>{partOfSpeech}</p>
        <div className={`${styles.definitionContainer} ${meanings.length > 1 && styles.multipleDefinitionContainer}`}>
            {
                meanings.map((definition,idx) =>
                    <div key={idx}>
                        { phonetic ? <p className={styles.phonetic}>{phonetic}</p> : null }
                        <p className={styles.definition}>{`${idx + 1}.- ${definition.definition}`}</p>
                        { definition.example ?
                            <>
                                <span className={styles.exampleTitle}>example: </span>
                                <span className={styles.example}>{definition.example}</span>
                            </> 
                            : null
                        }
                    </div>
                )
            }
        </div>
        </section>
    </>
);

export default DefinitionBox;