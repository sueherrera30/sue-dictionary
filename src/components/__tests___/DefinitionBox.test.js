import { render, screen, cleanup, fireEvent } from '@testing-library/react';

import DefinitionBox from '../DefinitionBox';


afterEach(() => {
    cleanup();
});

const mockdata = {
    meanings: [
       {
        definition: 'the one that you got birth',
        example: 'George is son of Martha' 
       }
    ],
    artOfSpeech: 'sustantive',
    phonetic: '/son/', 
}

test('<DefinitionBox />', () => {
    render(<DefinitionBox meanings={mockdata.meanings} partOfSpeech={mockdata.partOfSpeech} phonetic={mockdata.phonetic} />, );
    expect(screen.container).toMatchSnapshot();
});