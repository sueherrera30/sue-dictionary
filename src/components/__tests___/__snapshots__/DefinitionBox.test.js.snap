// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DefinitionBox /> 1`] = `undefined`;
